apps/omg_db/lib/db.ex

```diff
@@ -65,17 +63,41 @@ defmodule OMG.DB do
                       child_top_block_number: 1,
                       get_single_value: 2
 
-  def start_link(args), do: RocksDB.start_link(args)
+  @default_instance_name OMG.DB.Instance.Default
 
-  def child_spec(), do: RocksDB.child_spec()
-  def child_spec(args), do: RocksDB.child_spec(args)
+  def start_link(args) do
+    args
+    |> Keyword.put_new_lazy(:db_path, fn -> Application.fetch_env!(:omg_db, :path) end)
```

Comment:
True. Usually we try to pass params to process via `args`. However as an exception, base storage path for `omg_db` is initialized by release task `set_key_value_db` which puts it into app env variable. (as we discussed, omg_db needs to start earlier than Watcher, so it will be too late...)

Reply:
but you're already passing `db_path` sometimes:
eg. `OMG.DB.child_spec(db_path: Application.fetch_env!(:omg_db, :path), instance: OMG.DB.Instance.ExitProcessor),` in `OMG.Watcher.Supervisor`. 

I think it is better following that way? I feel the need of omg_db application does not restrict this.

Reply:
First thing - you comment is beautiful - really! I appreciate the effort. 

Second thing is unfortunately `:omg_db` is used by `:omg_eth` and `:omg` which are deps of watcher. So they need to start before Watcher, and `:omg_eth` even fetches something from db... It looks like `omg_db` is really deep hardwired 😭